src/internal/components/token-list/styles.scss

```diff
@@ -10,20 +10,34 @@
 .root {
   display: flex;
   flex-wrap: wrap;
-  flex-direction: column;
   gap: awsui.$space-xs;
+
+  &.horizontal {
+    flex-direction: row;
+  }
+  &.vertical {
+    flex-direction: column;
+  }
 }
 
 .list {
-  display: flex;
-  flex-wrap: wrap;
-  gap: awsui.$space-xs;
-}
-.list-horizontal {
-  flex-direction: row;
-}
-.list-vertical {
-  flex-direction: column;
+  display: contents;
```

Comment:
`display: contents` allows to omit the list from rendering so that the parent flex container styles apply. That makes list items and the following "show more" and "clear filters" buttons to be rendered in the same flex container which is required for the property-filter.

There is a risk that `display: contents` would also exclude the element from the a11y tree. My initial round of testing did not reveal any issues with it though.